Userland/Utilities/nc.cpp

```diff
@@ -144,16 +146,24 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         TRY(Core::System::setsockopt(fd, SOL_SOCKET, SO_RCVTIMEO, &timeout, sizeof(timeout)));
         TRY(Core::System::setsockopt(fd, SOL_SOCKET, SO_SNDTIMEO, &timeout, sizeof(timeout)));
 
-        auto* hostent = gethostbyname(target.characters());
-        if (!hostent) {
-            warnln("Socket::connect: Unable to resolve '{}'", target);
-            return 1;
-        }
-
         sockaddr_in dst_addr {};
         dst_addr.sin_family = AF_INET;
         dst_addr.sin_port = htons(port);
-        dst_addr.sin_addr.s_addr = *(in_addr_t const*)hostent->h_addr_list[0];
+
+        if (!numeric_mode) {
+            auto* hostent = gethostbyname(target.characters());
+            if (!hostent) {
+                warnln("Socket::connect: Unable to resolve '{}'", target);
+                return 1;
+            }
+            dst_addr.sin_addr.s_addr = *(in_addr_t const*)hostent->h_addr_list[0];
+        } else {
+            dst_addr.sin_addr.s_addr = htonl(INADDR_ANY);
```

Comment:
Isn't this just elaborate zeroing of the field? Do we need that at all, considering that `inet_pton` (at least if I understood correctly) will either set the address or exit the program?